### PR TITLE
CB-18143 CB-18144 add ccm flows to edh

### PR DESCRIPTION
**Pull request by doktoric**
See detailed description in the commit message.

### CODE REVIEW

**Review by topolyai5 — APPROVED** [review]

File: structuredevent-service-cdp/src/main/java/com/sequenceiq/cloudbreak/structuredevent/service/telemetry/mapper/FreeIpaUseCaseMapper.java [hunk 1]
```diff
@@ -79,6 +81,11 @@ private UsageProto.CDPFreeIPAStatus.Value lastStepToUseCaseMapping(String rootFl
                             UsageProto.CDPFreeIPAStatus.Value.UPGRADE_FINISHED,
                             UsageProto.CDPFreeIPAStatus.Value.UPGRADE_FAILED);
                     break;
+                case "UpgradeCcmFlowEventChainFactory":
+                    useCase = getFreeIpaStatus(nextFlowState, "UPGRADE_CCM_FINISHED_STATE_NAME",
```
Comment (nadamdb) [inline structuredevent-service-cdp/src/main/java/com/sequenceiq/cloudbreak/structuredevent/service/telemetry/mapper/FreeIpaUseCaseMapper.java:85]: Sanity check: was the intention here to create a CCM_UPGRADE_STARTED event when the UpgradeCcmFlow starts in the UpgradeCcmFlowEventChain and have a CCM_UPGRADE_FINISHED event when the UpgradeCcmFlow finishes and not when the whole flowchain finishes (so when UpdateUserDataFlow finishes)?

File: structuredevent-service-legacy/src/main/java/com/sequenceiq/cloudbreak/structuredevent/service/telemetry/mapper/ClusterUseCaseMapper.java [hunk 2]
```diff
@@ -143,6 +145,11 @@ private UsageProto.CDPClusterStatus.Value lastStepToUseCaseMapping(String rootFl
                     useCase = getClusterStatus(nextFlowState, "CERT_ROTATION_FINISHED_STATE", UsageProto.CDPClusterStatus.Value.RENEW_CLUSTER_INTERNAL_CERT_FINISHED,
                             UsageProto.CDPClusterStatus.Value.RENEW_CLUSTER_INTERNAL_CERT_FAILED);
                     break;
+                case "UpgradeCcmFlowEventChainFactory":
+                    useCase = getClusterStatus(nextFlowState, "UPGRADE_CCM_FINISHED_STATE",
```
Comment (nadamdb) [inline structuredevent-service-legacy/src/main/java/com/sequenceiq/cloudbreak/structuredevent/service/telemetry/mapper/ClusterUseCaseMapper.java:149]: Same question here as above

**Review by nadamdb — CHANGES_REQUESTED** [review]

File: structuredevent-service-cdp/src/main/java/com/sequenceiq/cloudbreak/structuredevent/service/telemetry/mapper/FreeIpaUseCaseMapper.java (hunk 1 above)
Comment (doktoric) [reply structuredevent-service-cdp/src/main/java/com/sequenceiq/cloudbreak/structuredevent/service/telemetry/mapper/FreeIpaUseCaseMapper.java:85]: done

File: structuredevent-service-legacy/src/main/java/com/sequenceiq/cloudbreak/structuredevent/service/telemetry/mapper/ClusterUseCaseMapper.java (hunk 2 above)
Comment (doktoric) [reply structuredevent-service-legacy/src/main/java/com/sequenceiq/cloudbreak/structuredevent/service/telemetry/mapper/ClusterUseCaseMapper.java]: done

**Review by nadamdb — APPROVED** [review]
LGTM

File: usage-collection/src/main/proto/usage.proto [hunk 3]
```diff
@@ -1588,8 +1618,8 @@ message CDPProxyDetails {
 
 // Show services for which the customer has provided their own DNS zones
 message CDPOwnDnsZones {
-  // Whether DNS is provided for the postgres by the customer or not
-  bool postgres = 1;
+    // Whether DNS is provided for the postgres by the customer or not
+    bool postgres = 1;
 }
```
Comment (lajosrodek) [inline usage-collection/src/main/proto/usage.proto:1623]: Not a big deal but think this extra indentation was not intentional. This file seem to use 2 spaces elsewhere. :smile:

File: usage-collection/src/main/proto/usage.proto [hunk 4]
```diff
@@ -3307,10 +3357,10 @@ message CDPStackPatchEvent {
...
   enum Value {
-    UNSET = 0;
-    AFFECTED = 1;
-    SUCCESS = 2;
-    FAILURE = 3;
+      UNSET = 0;
+      AFFECTED = 1;
+      SUCCESS = 2;
+      FAILURE = 3;
   }
```
Comment (lajosrodek) [inline usage-collection/src/main/proto/usage.proto:3364]: Like above, this indentation looks incorrect.

**Review by lajosrodek — COMMENTED** [review]
2 minor notes.

File: usage-collection/src/main/proto/usage.proto (hunk 3 above)
Comment (doktoric) [reply usage-collection/src/main/proto/usage.proto:1623]: this is copy paste from thunderhead

File: usage-collection/src/main/proto/usage.proto (hunk 4 above)
Comment (doktoric) [reply usage-collection/src/main/proto/usage.proto:3364]: this is copy paste from thunderhead

**Pull request merged**